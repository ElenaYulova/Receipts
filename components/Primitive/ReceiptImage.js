import React from 'react';
import PropTypes from 'prop-types';

import './ReceiptImage.css';

// Компонент img кулинарной книги. Переиспользуемый
export default class ReceiptImage extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string.isRequired,
        imageUrl: PropTypes.string.isRequired,
        
    };

    
    render() {
        const name = this.props.name;
        const imageUrl = this.props.imageUrl;
        return <div className="receipt-form-Image">

      <img src={imageUrl} onError={(EO)=>{EO.target.src='/Images/default.jpg'}} id={name}
        className="receipt-form-Image__Image"
        alt={name} title={name}/>
          
    </div>
 }   
}
