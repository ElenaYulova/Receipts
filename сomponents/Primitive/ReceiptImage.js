import React from 'react';
import PropTypes from 'prop-types';

import './ReceiptImage.css';

// Компонент img кулинарной книги. Переиспользуемый
export default class ReceiptImage extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string.isRequired,
        imageUrl: PropTypes.string.isRequired,
        
    };

    
    render() {
        const name = this.props.name;
    return <div className="receipt-form-Image">

      <img url={this.props.imageUrl || "../../Images/img.jpg"} id={name}
        className="receipt-form-Image__Image"
        alt={name} title={name}/>
          
    </div>
 }   
}
