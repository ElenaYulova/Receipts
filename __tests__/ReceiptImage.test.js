"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import ReceiptImage from '../components/Primitive/ReceiptImage';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('>>>ReceiptImage --- Shallow Render REACT COMPONENTS',()=>{
    //shapshots
       
      const receipt1Component = <ReceiptImage 
      name = "name1"
      imageUrl = "../../Images/salad-chick-q.jpg"
       />;
    
       const receipt2Component = <ReceiptImage 
       name = "name2"
      imageUrl = "2222"
        />;
    
      
    
         const component1 = renderer.create(receipt1Component);
    
         it('renders usual receipt', () => {
           let componentTree = component1.toJSON();
           expect(componentTree).toMatchSnapshot();
         });
    
         const component2 = renderer.create(receipt2Component);
    
        it('renders receipt with wrong url', () => {
            let componentTree = component2.toJSON();
            expect(componentTree).toMatchSnapshot();
        });
    
    });