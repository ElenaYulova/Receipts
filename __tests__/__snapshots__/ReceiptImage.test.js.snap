// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ReceiptImage --- Shallow Render REACT COMPONENTS renders receipt with wrong url 1`] = `
<div
  className="receipt-form-Image"
>
  <img
    alt="name2"
    className="receipt-form-Image__Image"
    id="name2"
    onError={[Function]}
    src="2222"
    title="name2"
  />
</div>
`;

exports[`>>>ReceiptImage --- Shallow Render REACT COMPONENTS renders usual receipt 1`] = `
<div
  className="receipt-form-Image"
>
  <img
    alt="name1"
    className="receipt-form-Image__Image"
    id="name1"
    onError={[Function]}
    src="../../Images/salad-chick-q.jpg"
    title="name1"
  />
</div>
`;
